### DISCUSSION TITLE
Installation problems on Mac

### DISCUSSION THREAD

**Discussion by robl01** [James-Yu/LaTeX-Workshop#3459] [Q&A] [answered]
## Preliminary questions 

I tried to install Latex some months ago, but it did not work and I wanted to try again. By googling the issue, I got to a report but I could not understand the solutions (I am "just a mechanical engineer" with very few computer knowledge 🥲 )

### Disable all the other extensions except for LaTeX Workshop, restart VS Code, and check that you still see this issue. [Required]

You still see this issue?: Yes

### If your issue is with compiling a document (not having to do with finding the root file of a project), check first that you can compile manually.

The document is a sample document I found online. It should work :) 

## Describe the bug [Required]

I wanted to run the LaTeX script. By clicking on the "run button" in the top right corner in VSCode, I got this error: 

Recipe terminated with fatal error: spawn latexmk ENOENT.

## Logs [Required]

The Log message is the following:

[16:38:31] BUILD command invoked.
[16:38:31] The document of the active editor: file:///Users/robert/Desktop/Test.tex
[16:38:31] The languageId of the document: latex
[16:38:31] Current workspace folders: undefined
[16:38:31] Found root file from active editor: /Users/robert/Desktop/Test.tex
[16:38:31] Keep using the same root file: /Users/robert/Desktop/Test.tex
[16:38:31] Building root file: /Users/robert/Desktop/Test.tex
[16:38:31] Build root file /Users/robert/Desktop/Test.tex
[16:38:31] outDir: /Users/robert/Desktop
[16:38:31] Recipe step 1: latexmk
[16:38:31] Recipe step 1 args: ["-synctex=1","-interaction=nonstopmode","-file-line-error","-pdf","-outdir=/Users/robertblaszczyk/Desktop","/Users/robert/Desktop/Test"]
[16:38:31] Recipe step env: {}
[16:38:31] cwd: /Users/robert/Desktop
[16:38:31] LaTeX build process spawned. PID: undefined.
[16:38:31] LaTeX fatal error: spawn latexmk ENOENT, . PID: undefined.
[16:38:31] Does the executable exist? $PATH: /opt/homebrew/Caskroom/miniforge/base/bin:/opt/homebrew/Caskroom/miniforge/base/condabin:opt/homebrew/bin:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin:/opt/homebrew/bin:/opt/homebrew/bin
[16:38:31] Does the executable exist? $Path: undefined
[16:38:31] The environment variable $SHELL: /bin/zsh

## Desktop [Required]

Please write exact version numbers. Please don't write `latest` instead of exact numbers.

 - OS: Mac OS 12.2.1
 - VS Code version: 1.71.0
 - Extension version: 8.29.0
 - TeX distribution version: Here I am not sure: I downloaded MacTex a while ago, but it did not work properly. This may be the issue I cannot fix.

If you need any additional information let me know, I will try to provide them. Thank you already in advance for your help

**Comment by we-taper** [comment]
I am having a similar issue as well. In my case, that the environment variable `PATH` loaded by this extension is different from the `PATH` loaded by vscode. In vscode's terminal, it could load the `PATH` correctly and found the executable (in my case `pdflatex`), but inside LaTeX-Workshop, the `PATH` is different and does not include the right executables for latex.

Could you try `which latexmk` to find out where it is installed?

**Reply by James-Yu** [reply]
> Could you try `which latexmk` to find out where it is installed?

No, this is not a viable solution, as in such case `latexmk` is not in PATH and cannot be found even `which latexmk`ed.

**Reply by we-taper** [reply]
Thanks @James-Yu  for the prompt reply. Let me explain in more detail. I believe my `PATH` has been setup to include the right executables, but it is not properly respected when the extension tries to spawn processes. 

Assuming the latex is installed via `brew`, which in my case setup the `PATH` properly. `latexmk` is inside the `PATH` and could be found by `which latexmk`, even in VSCode's terminal.

However, my LaTeX-Workshop extension still reports missing that executable. And the `PATH` report by LaTeX-Workshop is different from the `PATH` available from my terminal (whether inside or outside VSCode).

Since my the `PATH` setup by `/usr/libexec/path_helper` properly includes `pdflatex`, it could be that LaTeX-Workshop (or VSCode's extension in general? I am not sure) does not respect the `path_helper` mechanism in MacOS.

BTW, this [issue #13701](https://github.com/microsoft/vscode/issues/13701) in VSCode talks about setting up `path_helper` in the VSCode terminal. It is related but does not answer our problem.

**Reply by James-Yu** [reply]
It's VSCode. This extension does not and cannot change the shell env that VSCode resides. The extension can only read PATH from it.

**Reply by we-taper** [reply]
Thanks. I am asking VSCode to see if they can find out the reason here at https://github.com/microsoft/vscode/issues/173480

**Comment by if-david** [comment, marked as answer]
Don't waste any more time—after trying at least 10 different solutions, the only one that works is installing the Full MacTeX...
